mlir/lib/Driver/CompilerDriver.cpp

```diff
@@ -142,13 +151,19 @@ void registerAllCatalystDialects(DialectRegistry &registry)
 FailureOr<llvm::Function *> getJITFunction(MLIRContext *ctx, llvm::Module &llvmModule)
 {
     Location loc = NameLoc::get(StringAttr::get(ctx, llvmModule.getName()));
+    std::list<StringRef> visited;
     for (auto &function : llvmModule.functions()) {
-        emitRemark(loc) << "Found LLVM function: " << function.getName() << "\n";
+        visited.push_back(function.getName());
         if (function.getName().starts_with("catalyst.entry_point")) {
             return &function;
         }
     }
-    emitError(loc, "Failed to find JIT function");
+    withoutStackTrace(ctx, [&]() {
+        auto noteStream =
+            emitRemark(loc, "Failed to find entry-point function among the following: ");
+        llvm::interleaveComma(visited, noteStream, [&](StringRef t) { noteStream << t; });
+    });
```

Comment:
Maybe we should also define an entry point that is not a variable name? That would simplify some parts of the frontend as well I think. (But that does not belong in this ticket).

Reply:
How does the stack trace look like in this example? Is it too overwhelming for the user?

Reply:
@erick-xanadu here is the trace (the same as in the other reply below) https://gist.github.com/grwlf/ff5040b783bcb2edabd561b99a98acbf  Note that there are no errors, just remarks.

Reply:
>  it just says "couldn't find foo" 

I think it is fine in printing some details. My intention here is to preserve the same information that compiler driver sends to user, but maybe reshape it into a more compact form. Note that I removed the line-by-line printing of these function names. For me it is again a question of contract between Catalyst and users.

Reply:
I think the remark was there just for debugging. It shouldn't have been there in the first place. But since we don't have a lot of use for running from IR, it went unnoticed. I think a short message is fine :)

Reply:
Printing the functions is fine here I think. Looking at this more closely with Sergei, this message is actually important, but when it is being printed (and for that matter when these lines are run) is not set up correctly. Sergei will open an issue to fix that in the future.

In summary, this code should only run when we are compiling from llvmir. The code is not relevant when compiling a Python function though.

Reply:
Ok, I will resolve this ticket then :)